ShopApp/ShopApp/Controller/ProductDetailsViewController.swift

```diff
@@ -1,300 +1,259 @@
+                                                          message: messagge,
+                                                          preferredStyle: .alert)
+                 allFieldRequired.addAction(UIAlertAction(title: String(localized: "Ok", comment: "Ok"), style: .cancel, handler: nil))
+                 self.present(allFieldRequired, animated: true, completion: nil)
+             }
+         }
+     }
+
+     // MARK: - Size button select actoin
+     @IBAction private func sizeSelect(sender: UIButton) {
+         guard sender.tintColor != UIColor.systemGray3 else {
+             return
+         }
+
+         for i in 0...5 {
+             sizeButtons[i].backgroundColor = .systemGray6
+         }
+
+         sender.backgroundColor = .systemGray3
+         guard let name = sender.titleLabel?.text as? String else {
+             return
+         }
+
+         self.sizeSelected = name
+     }
+
+     func favoritesFunction(complition: @escaping (String) -> Void) {
+         guard let email = Auth.auth().currentUser?.email as? String else { return }
+         let ref = Database.database().reference()
+
+         let emailRef = email.replacingOccurrences(of: ".", with: " ")
+         ref.child("users/\(emailRef)/favourites/\(self.product.getName())").observeSingleEvent(of: DataEventType.value, with: { snapshot in
+
+             guard let data = snapshot.value as? String else {
+                 print("there is not favrite item ")
+                 Database.database().reference().child("users/\(emailRef)/favourites/\(self.product.getName())").setValue("favorite")
+                 self.likeButton.setImage(UIImage(systemName: "hand.thumbsup.fill"), for: .normal)
+                 complition("added")
+                 return
+             }
+
+             Database.database().reference().child("users/\(emailRef)/favourites/\(self.product.getName())").removeValue()
+             self.likeButton.setImage(UIImage(systemName: "hand.thumbsup"), for: .normal)
+             complition("removed")
+         })
+     }
+
+     // MARK: - Set the like button in the beginning
+     func setFavoritesButton(complition: @escaping (String) -> Void) {
+         guard let email = Auth.auth().currentUser?.email as? String else { return }
+         let ref = Database.database().reference()
+
+         let emailRef = email.replacingOccurrences(of: ".", with: " ")
+         ref.child("users/\(emailRef)/favourites/\(self.product.getName())").observeSingleEvent(of: DataEventType.value, with: { snapshot in
+
+             guard let data = snapshot.value as? String else {
+                 self.likeButton.setImage(UIImage(systemName: "hand.thumbsup"), for: .normal)
+                 complition("added")
+                 return
+             }
+
+             self.likeButton.setImage(UIImage(systemName: "hand.thumbsup.fill"), for: .normal)
+             complition("removed")
+         })
+     }
```

Comment:
This method is pretty similar to **favoritesFunction**, maybe you can creat just one